javaagent-extensions/gcp-auth/README.md

```diff
@@ -0,0 +1,69 @@
+
+The Google Cloud Auth Extension allows the users to export telemetry from their applications auto-instrumented using the OpenTelemetry Java Agent to Google Cloud using the built-in OTLP exporters.
+The extension takes care of the necessary configuration required to authenticate to GCP to successfully export telemetry.
+
+## Prerequisites
+
+### Ensure the presence of Google Cloud Credentials on your machine/environment
+
+```shell
+gcloud auth application-default login
+```
+Executing this command will save your application credentials to default path which will depend on the type of machine -
+- Linux, macOS: `$HOME/.config/gcloud/application_default_credentials.json`
+- Windows: `%APPDATA%\gcloud\application_default_credentials.json`
+
+**NOTE: This method of authentication is not recommended for production environments.**
+
+Next, export the credentials to `GOOGLE_APPLICATION_CREDENTIALS` environment variable -
+
+For Linux & MacOS:
+```shell
+export GOOGLE_APPLICATION_CREDENTIALS=$HOME/.config/gcloud/application_default_credentials.json
+```
+
+These credentials are built-in running in a Google App Engine, Google Cloud Shell or Google Compute Engine environment.
+
+### Configuring the extension
+
+The extension can be configured either by environment variables or system properties.
+
+Here is a list of configurable options for the extension:
+
+ - `GOOGLE_CLOUD_PROJECT`: Environment variable that represents the Google Cloud Project ID to which the telemetry needs to be exported.
+   - Can also be configured using `google.cloud.project` system property.
+
+It is **mandatory** to configure this option for the extension, not doing so would lead to a `ConfigurationException`.
+
+## Usage
+
+The OpenTelemetry Java Agent Extension can be easily added to any Java application by modifying the startup command to the application.
+For more information on Extensions, see the [documentation here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/README.md).
+
+Below is a snippet showing how to add the extension to a Java application using the Gradle build system.
+
+```gradle
+// Specify OpenTelemetry Autoinstrumentation Java Agent Path.
+def otelAgentPath = <OpenTelemetry Java Agent location>
+// Specify the path for Google Cloud Authentication Extension for the Java Agent.
+def extensionPath = <Google Cloud Authentication Extension location>
+def googleCloudProjectId = <Your Google Cloud Project ID>
+def googleCloudRegion = <Your preferred Google Cloud Region>
+def googleOtlpEndpoint = <Google Cloud OTLP endpoint>
+
+application {
+   ...  
+  "-javaagent:${otelAgentPath}",
+  "-Dotel.javaagent.extensions=${extensionPath}",
+  // Configure the GCP Auth extension using system properties.
+  // This can also be configured using environment variables.
+  "-Dgoogle.cloud.project=${googleCloudProjectId}",
+  "-Dgoogle.cloud.region=${googleCloudRegion}",  
```

Comment:
do you use this for anything? I don't see it anywhere

Reply:
This is how you would configure the extension. I have a sample ready for this which I haven't pushed yet, this config is used there.

You can also see this config being used in the `ExtensionIntegrationTest.java`.

Reply:
I still don't see google.cloud.region being used anywhere...

Reply:
Ah, yes - that was removed since we now allow the OTLP endpoint to be defined by the user. It was earlier used to construct the regionalized endpoint.